hack/kind-up.sh

```diff
@@ -149,6 +150,28 @@ server = "${UPSTREAM_SERVER}"
 EOF
 }
 
+check_registry_cache_availability() {
+  if [[ "$REGISTRY_CACHE" != "true" ]]; then
+    return
+  fi
+  echo "Registry-cache enabled. Checking if registry-cache instances are deployed in prow cluster."
+  check_registry_cache_dns_entry registry-gcr-io
+  check_registry_cache_dns_entry registry-eu-gcr-io
+  check_registry_cache_dns_entry registry-ghcr-io
+  check_registry_cache_dns_entry registry-registry-k8s-io
+  check_registry_cache_dns_entry registry-quay-io
+}
+
+check_registry_cache_dns_entry() {
+  local name=$1
+  local registry_cache_ip
+  registry_cache_ip=$(getent hosts "$name.kube-system.svc.cluster.local" | awk '{ print $1 }')
+  if [[ "$registry_cache_ip" == "" ]]; then
+    echo "Unable to resolve IP of $name.kube-system.svc.cluster.local in prow cluster. Disabling registry-cache."
+    REGISTRY_CACHE=false
+  fi
+}
```

Comment:
While I understand the motivation behind this - do we really need this? We are repeating all registries again here. It can become a bit tedious when somebody adds a new registry and needs to find all the places that have to be adapted.
Alternatively, can the registry list in L158-L162 be read from somewhere instead of getting hard-coded there?

Reply:
The script is now running a `kubectl create ... --dry-run=client ...` for kustomize of `registry-prow`  and checks DNS for all lines which contain `kube-system.svc.cluster.local`